pkg/controller/deployment/recreate.go

```diff
@@ -99,9 +102,27 @@ func (dc *DeploymentController) scaleDownOldReplicaSetsForRecreate(oldRSs []*ext
 func (dc *DeploymentController) waitForInactiveReplicaSets(oldRSs []*extensions.ReplicaSet) error {
 	for i := range oldRSs {
 		rs := oldRSs[i]
-
-		condition := rsutil.ReplicaSetIsInactive(dc.client.Extensions(), rs)
-		if err := wait.ExponentialBackoff(unversionedclient.DefaultRetry, condition); err != nil {
+		desiredGeneration := rs.Generation
+		observedGeneration := rs.Status.ObservedGeneration
+		specReplicas := rs.Spec.Replicas
+		statusReplicas := rs.Status.Replicas
+
+		if err := wait.ExponentialBackoff(unversionedclient.DefaultRetry, func() (bool, error) {
+			replicaSet, err := dc.rsStore.ReplicaSets(rs.Namespace).Get(rs.Name)
+			if err != nil {
+				return false, err
+			}
+
+			specReplicas = replicaSet.Spec.Replicas
```

Comment:
No, status.replicas=0 doesn't guarantee spec.replicas=0. Actually that means that the replica set controller is about to create new pods.